reactDashboard/src/chartjs-plugin-annotation/types/line.js

```diff
@@ -0,0 +1,448 @@
+// Line Annotation implementation
+module.exports = function (Chart) {
+  /* eslint-disable global-require */
+  var chartHelpers = Chart.helpers;
+  var helpers = require("../helpers.js")(Chart);
+  /* eslint-enable global-require */
+
+  var horizontalKeyword = "horizontal";
+  var verticalKeyword = "vertical";
+
+  function LineFunction(view) {
+    // Describe the line in slope-intercept form (y = mx + b).
+    // Note that the axes are rotated 90° CCW, which causes the
+    // x- and y-axes to be swapped.
+    var m = (view.x2 - view.x1) / (view.y2 - view.y1);
+    var b = view.x1 || 0;
+
+    this.m = m;
+    this.b = b;
+
+    this.getX = function (y) {
+      // Coordinates are relative to the origin of the canvas
+      return m * (y - view.y1) + b;
+    };
+
+    this.getY = function (x) {
+      return (x - b) / m + view.y1;
+    };
+
+    this.intersects = function (x, y, epsilon) {
+      epsilon = epsilon || 0.001;
+      var dy = this.getY(x);
+      var dx = this.getX(y);
+      return (
+        (!isFinite(dy) || Math.abs(y - dy) < epsilon) &&
+        (!isFinite(dx) || Math.abs(x - dx) < epsilon)
+      );
+    };
+  }
+
+  function calculateLabelPosition(
+    view,
+    width,
+    height,
+    padWidth,
+    padHeight,
+    x1,
+    x2,
+    y1,
+    y2
+  ) {
+    var line = view.line;
+    var ret = {};
+    var xa = 0;
+    var ya = 0;
+
+    switch (true) {
+      // top align
+      case view.mode === verticalKeyword && view.labelPosition === "top":
+        ya = padHeight + view.labelYAdjust;
+        xa = width / 2 + view.labelXAdjust;
+        ret.y = y1 + ya;
+        ret.x = (isFinite(line.m) ? line.getX(ret.y) : x1) - xa;
+        break;
+
+      // top align fitted within chart area
+      case view.mode === verticalKeyword && view.labelPosition === "top-fitted":
```

Comment:
This case block is the relevant addition. It calculates the bounds as in the case block above, then adjusts the label to not go off the left or right edge of the chart. If the line being labelled goes off the edge of the chart, the label is also moved out of the visible area to avoid seeing a partial label without a line visible.

Reply:
In my testing, I was unable to have the label appear at the earliest **possible** date in the graph (`2020-02-28`). The label appears if set to `2020-02-29`.
Adjusting`xPadding: 10` to 1 didn't resolve this.

Reply:
Yup - good catch. Resolved now